@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TextElementType } from "./TextElementType";
+
+/**
+ * State of fading.
+ */
+export enum FadingState {
+    Undefined = 0,
+    FadingIn = 1,
+    FadedIn = 2,
+    FadingOut = -1,
+    FadedOut = -2
+}
+
+/**
+ * Time to fade in/fade out the labels in milliseconds.
+ */
+export const DEFAULT_FADE_TIME = 800;
+
+/**
+ * State of rendering of the icon and text part of the `TextElement`. Mainly for fading the elements
+ * in and out, to compute the opacity.
+ *
+ * @hidden
+ */
+export class RenderState {
```

Comment:
I think this should be split into a separate file, you have already created a separate file for TextElementType, so another would not hurt.